### PR TITLE
Killian/kafka

### PR DESCRIPTION
**Pull request by killianC2323**
Added Kafka producers for both posts and profile microservices that produce messages to "post-updates" topic and "profile-updates" topic. These messages are then consumed by the notification microservice Kafka consumer.

### CODE REVIEW

**Review by M-Byte480 — CHANGES_REQUESTED** [review]
In the NotificationService.java can you change the @Autowired to:
```java
    private final NotificationRepository notificationRepository;
    private final NotificationMapper notificationMapper;

    public NotificationService(NotificationRepository notificationRepository, NotificationMapper notificationMapper) {
        this.notificationRepository = notificationRepository;
        this.notificationMapper = notificationMapper;
    }```

File: .env.example
Comment (Silver-Golden) [inline .env.example:1]: Out of curiousity why is this deleted?  
By rights we should be expanding it isntead

File: .gitignore [hunk 1]
```diff
@@ -42,8 +42,8 @@ test.*
 .attach_pid*
 
 # Environment Variables
-!.env.example
-.env*
```
Comment (Silver-Golden) [inline .gitignore:46]: Do not change this, especially not to this.

L45 is now saying "do not ignore any ``.env``"  
L46 is now saying "Ignore any ``.env``"  
So you have made a contradiction

This relates to why did you delete the ``.env.example``

**Review by Silver-Golden — CHANGES_REQUESTED** [review]
Overall looks good.

The env stuff needs to be explained and then fixed (to what it was)  

Also beware of any unintended whitespace changes (you have them in a few files)

File: .gitignore (hunk 1 above)
Comment (killianC2323) [reply .gitignore:46]: didn't even realise I changed them sorry, will fix it now

File: Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java [hunk 2]
```diff
@@ -25,13 +26,17 @@ public class ControllerProfile {
     // Bean injections
     // It is okay to have more than one service, but nothing lower
     private final ProfileService profileService;
+    private final KafkaProducer kafkaProducer;
 
-    public ControllerProfile(ProfileService profileService) {
+    public ControllerProfile(ProfileService profileService, KafkaProducer kafkaProducer) {
+        this.kafkaProducer = kafkaProducer;
         this.profileService = profileService;
+
```
Comment (Bourkupine) [inline Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java:34]: dont need this extra line

**Review by Bourkupine — COMMENTED** [review]
Looks good
